Composer/plugins/azurePublish/src/node/azureProvisionUtil.ts

```diff
@@ -0,0 +1,323 @@
+                this.logger({
+                    status: AzureAPIStatus.ERROR,
+                    message: resourceGroupsResult._response.bodyAsText
+                });
+                return []
+            }
+            return resourceGroupsResult._response.parsedBody;
+        }
+        catch (err) {
+            this.logger({
+                status: AzureAPIStatus.ERROR,
+                message: JSON.stringify(err, Object.getOwnPropertyNames(err)),
+            });
+            return []
+        }
+    }
+
+    /**
+     * Check bot channel registration name availability
+     */
+    public async CheckBotNameAvailability(botName: string, subscriptionId: string): Promise<CheckNameAvailabilityResponseBody> {
+        try {
+            if (!botName) {
+                this.logger({
+                    status: AzureAPIStatus.PARAM_ERROR,
+                    message: 'Need bot name as a parameter.'
+                });
+                return {
+                    valid: false,
+                    message: 'Invalid param: bot name'
+                } as CheckNameAvailabilityResponseBody;
+            }
+            if (!subscriptionId) {
+                this.logger({
+                    status: AzureAPIStatus.PARAM_ERROR,
+                    message: 'Need subscription id as a parameter.'
+                });
+                return {
+                    valid: false,
+                    message: 'Invalid param: subscription id'
+                } as CheckNameAvailabilityResponseBody;
+            }
+
+            const azureBotService = new AzureBotService(this.credentials, subscriptionId);
+            const getCheckNameAvailabilityResult = await azureBotService.bots.getCheckNameAvailability({
+                name: botName,
+                type: 'bot'
+            });
+            if (getCheckNameAvailabilityResult._response.status >= 300) {
+                this.logger({
+                    status: AzureAPIStatus.ERROR,
+                    message: getCheckNameAvailabilityResult._response.bodyAsText
+                });
+                return {
+                    valid: false,
+                    message: `Invalid request: ${getCheckNameAvailabilityResult._response.bodyAsText}`
+                } as CheckNameAvailabilityResponseBody;
+            }
+            return getCheckNameAvailabilityResult._response.parsedBody;
+        }
+        catch (err) {
+            this.logger({
+                status: AzureAPIStatus.ERROR,
+                message: JSON.stringify(err, Object.getOwnPropertyNames(err)),
+            });
+            return {
+                valid: false,
+                message: JSON.stringify(err, Object.getOwnPropertyNames(err))
+            } as CheckNameAvailabilityResponseBody;
+        }
+    }
+
+    /**
+     * Check the web app name availability
+     */
+    public async CheckWebAppNameAvailability(webAppName: string, subscriptionId: string): Promise<ResourceNameAvailability> {
```

Comment:
If this is going to be used by the client, why is it here instead of along side the other /azure/ APIS in the provision controller?

Reply:
> If this is going to be used by the client, why is it here instead of along side the other /azure/ APIS in the provision controller?

I will move the file to the server/controller